tests/unittests/aggregation_tests.py

```
@@ -1615,6 +1616,354 @@ def test_expr(self):
+        )
+        self.assertEqual(result[0]["count"], 2)
+
+
+class HintDecisionTests(unittest.TestCase):
+    """Tests for the hint logic used in _run_parallel_consolidated.
+
+    The rule: hint={"_id": 1} is included only when the consolidated
+    pipeline has no $match stages that might benefit from a different index.
+    """
+
+    def _should_hint(self, pipeline):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_should_hint`

(ANN202)